### ISSUE TITLE
Using install_github() from behind a proxy with github enterprise

### ISSUE THREAD

**Issue by Mullefa** [hadley/devtools#688]
I have closed issue #656 as `install_github()` now has full functionality from behind a proxy (with public github).

 However, if after setting the environment variable `GITHUB_PAT` I try:

```R
devtools::install_github("guy.dawson/<my-package>", host = "github.<company>.com/api/v3")
```

(with or without the proxy set) I get - after about a minutes wait - the following error:

```
Proxy CONNECT aborted
```

**Comment by hadley** [comment]
I have no idea :(  I'd recommend trying with `httr::with_verbose()` and verifying that you can submit regular requests through your proxy .

**Comment by Mullefa** [comment]
Here is the issue:

I'm on a server which is on the same network as my company github. This means that to install a package `P` from my company github using `install_github()` a proxy __can't__ be set.

However, lets say the packages (on CRAN) for which `P` depends are not installed. Then the call to `install_github()` will fail unless I set the argument `dependencies = TRUE`. However, with `dependencies = TRUE`, this results to a call to `install.packages()` which will hang because a proxy hasn't been set.

The situation is even worse because of the following: 

Say I have started an R session on the server where a proxy hasn't been set in the parent environment. Then calls to `install.packages(...)` and `install_github(..., host = "api.github.com")` will hang.

 If I use httr from within R to set the proxy then calls to `install_github(..., host = "api.github.com")` will now work, however, calls to `install.packages(...)` will still hang - presumably because `install.packages()` isn't using the functions in httr. 

As an initial step towards this problem, I think it would be good to create a new function `install_package()` - like `install.packages()`, but uses the functions in `httr` to download the relevant source code from CRAN.

As a second step, it would be great to have an option to pass `httr::use_proxy()` to the install functions in devtools, so that the proxy is set at the function level. 

Ultimately, I would love a package which can replicate the behaviour of pip for python. To this end I have set up a spike called [installr](https://github.com/Mullefa/installr).

Cheers, and keep up the good work :)
